sep2/immer-yjs
# ci: init automated checks

Resolves #14

Adding automated checks:

For the pull requests:
- Title check: checks that the title it formatted according to the [conventional commits specs](https://www.conventionalcommits.org/en/v1.0.0/). If we would like to ensure that all commit messages are formatted as such, we could configure the repository so that only squash and merge is allowed. In that way, the title from the PR will become the commit message.

For the source code:
- Code formatting: note that this job will be failing in this PR. This is ok, since I'll address this in a separate PR.
- Type checking
- Linter
- Tests
- Build. Not very useful on its own, but the idea is that the output will be tested in another step (#18). This will require a separate PR, which is why tha

@sep2 Once this PR has been merged, I'd like to request a branch protection rule for main that requires both of these checks to pass.



## How to test

Create a new branch based on this one. Make changes, open a stacked PR, and observe the result from the automated checks:

Check that all jobs always run, even if the previous jobs failed. For example, if the code formatter fails, we still want to see if the linter fails.
